### ISSUE TITLE
Dist::Zilla::App::Tester kills refs in @INC

### ISSUE THREAD

**Issue by kentfredric** [rjbs/Dist-Zilla#441]
```perl
local @INC = map {; File::Spec->rel2abs($_) } @INC;
```

I was wondering why neither `Devel::Hide` or `Devel::TraceUse` could tell me where the Moose dependency was coming from till I found this :/ 

```
 perl -MDevel::Hide=Moose -MFile::Spec -MData::Dump=pp -E '@INC = map {; File::Spec->rel2abs($_) } @INC; pp \@INC'
Devel::Hide hides Moose.pm
[
  "/home/kent/perl/kentnl/Dist-Zilla-App-Command-dumpphases/CODE(0xcdf488)",
  "/home/kent/perl5/perlbrew/perls/5.21.8-c/lib/site_perl/5.21.9/x86_64-linux",
  "/home/kent/perl5/perlbrew/perls/5.21.8-c/lib/site_perl/5.21.9",
  "/home/kent/perl5/perlbrew/perls/5.21.8-c/lib/5.21.9/x86_64-linux",
  "/home/kent/perl5/perlbrew/perls/5.21.8-c/lib/5.21.9",
  "/home/kent/perl/kentnl/Dist-Zilla-App-Command-dumpphases",
]
```

Yeah. File::Spec's rel2abs blindly stringifies those references.

**Comment by rjbs** [comment]
The map here should just pass references through unchanged.

**Comment by karenetheridge** [comment]
So is this a bug with File::Spec, or Dist::Zilla?

**Comment by kentfredric** [comment]
I would be inclined to say "Both", but more dzil than File::Spec. Stringification by default makes sense for File::Spec due to Path::(Class|Tiny) existing. But you cant use those in `@INC`, and so any refs in `@INC` must be assumed to be `@INC` loaders.

**Comment by rjbs** [comment]
Thanks, fixed in 302cb90.
